crates/red_knot_python_semantic/src/types/class.rs

```diff
@@ -1167,6 +1181,11 @@ impl<'db> KnownClass {
                     KnownModule::Builtins
                 }
             }
+            Self::NotImplementedType => {
+                // TODO: Currently, types.NotImplementedType is a type alias of `builtins._NotImplementedType`
+                // TypeAlias is not fully supported yet, so we use `_NotImplementedType` here for now
```

Comment:
I don't think we need this TODO comment. Even on newer Python versions, I think the simplest approach is to always consider `builtins._NotImplementedType` as the canonical definition of this type, since it's the one that will always exist.